tokio/src/sync/mutex.rs

```diff
@@ -301,6 +301,41 @@ impl<T: ?Sized> Mutex<T> {
         MutexGuard { lock: self }
     }
 
+    /// Blocking lock this mutex. When the lock has been acquired, function returns a
+    /// [`MutexGuard`].
+    ///
+    /// This method is intended for use cases where you
+    /// need to use this mutex in asynchronous code as well as in synchronous code.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::thread;
+    /// use std::sync::Arc;
+    /// use tokio::sync::Mutex;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let mutex =  Arc::new(Mutex::new(1));
+    ///
+    ///     let mutex1 = Arc::clone(&mutex);
+    ///     let sync_code = thread::spawn(move || {
+    ///         let mut n = mutex1.blocking_lock();
+    ///         *n = 2;
+    ///     });
+    ///
+    ///     sync_code.join().unwrap();
```

Comment:
It is generally frowned upon to use `JoinHandle::join` in async code because it is a blocking function. One alternative is to use `spawn_blocking`, which has a non-blocking join.
```suggestion
    ///     let sync_code = tokio::task::spawn_blocking(move || {
    ///         let mut n = mutex1.blocking_lock();
    ///         *n = 2;
    ///     });
    ///
    ///     sync_code.await.unwrap();
```

Reply:
Addressed! Thanks for your review!